libs/providers/config-cat/src/lib/config-cat-provider.ts

```diff
@@ -1,36 +1,52 @@
 import {
   EvaluationContext,
   JsonValue,
+  OpenFeatureEventEmitter,
   ParseError,
   Provider,
+  ProviderEvents,
   ResolutionDetails,
   ResolutionReason,
   StandardResolutionReasons,
   TypeMismatchError,
 } from '@openfeature/js-sdk';
-import * as configcat from 'configcat-js';
-import { IConfigCatClient } from 'configcat-js';
-import { SettingValue } from 'configcat-common/lib/RolloutEvaluator';
+import { getClient, IConfigCatClient, IEvaluationDetails, SettingValue } from 'configcat-js';
 import { transformContext } from './context-transformer';
 
 export class ConfigCatProvider implements Provider {
   private client: IConfigCatClient;
+  public events = new OpenFeatureEventEmitter();
+
+  public metadata = {
+    name: ConfigCatProvider.name,
+  };
 
   private constructor(client: IConfigCatClient) {
     this.client = client;
+
+    client.on('clientReady', () => this.events.emit(ProviderEvents.Ready));
+    client.on('configChanged', (projectConfig) =>
+      this.events.emit(ProviderEvents.ConfigurationChanged, { metadata: { ...projectConfig } })
+    );
+    client.on('clientError', (message: string, error) =>
+      this.events.emit(ProviderEvents.Error, {
+        message: message,
+        metadata: error,
+      })
+    );
```

Comment:
Could you move this to the initialize method?

https://open-feature.github.io/js-sdk/interfaces/OpenFeature_JS_SDK.Provider.html#initialize